src/data-workspace/min-max-limits-mutations/use-update-limits.test.js

```diff
@@ -0,0 +1,157 @@
+import { act, renderHook } from '@testing-library/react-hooks'
+import React from 'react'
+import { useDataValueSetQueryKey } from '../../shared/index.js'
+import { dataValueSets } from '../../shared/use-data-value-set/query-key-factory.js'
+import { Wrapper, useTestQueryClient } from '../../test-utils/index.js'
+import useUpdateLimits from './use-update-limits.js'
+
+jest.mock('../../shared/use-data-value-set/use-data-value-set-query-key.js', () => ({
+    __esModule: true,
+    default: jest.fn()
+}))
+
+describe('useUpdateLimits', () => {
+    useDataValueSetQueryKey.mockImplementation(() => dataValueSets.byIds({
+        dataSetId: 'ds-id',
+        periodId: 2022,
+        orgUnitId: 'ou-id',
+        categoryComboId: 'cc-id',
+        categoryOptionIds: ['co-id1', 'co-id2'],
+    }))
+
+    function createQueryClient() {
+        // There is no other way of getting the query client right now.
+        // The reason is: There is no way to manualy create a new data engine
+        // instance (from @dhis2/app-runtime) because the class is not being
+        // exported.
+        const { result: { current: queryClient } } = renderHook(useTestQueryClient)
+        return queryClient
+    }
+
+    it('should optimistically update the limits in the cache', async () => {
+        const queryClient = createQueryClient()
+        const { result, waitFor } = renderHook(useUpdateLimits, {
+            wrapper: ({ children }) => (
+                <Wrapper
+                    queryClient={queryClient}
+                    dataForCustomProvider={{
+                        'dataEntry/minMaxValues': { min: 3, max: 4 },
+                    }}
+                >
+                    {children}
+                </Wrapper>
+            ),
+        })
+
+        act(() => {
+            result.current.mutate({
+                categoryOptionCombo: 'category-option-combo-id',
+                dataElement: 'data-element-id',
+                orgUnit: 'orgUnitId',
+                minValue: 2,
+                maxValue: 3,
+            })
+        })
+
+        await waitFor(() => result.current.isSuccess)
+
+        const { queries } = queryClient.queryCache
```

Comment:
The `QueryCache`-class has a find method where you can provide the queryKey directly. See https://tanstack.com/query/v4/docs/react/reference/QueryCache#querycachefind

Reply:
Thanks @Birkbjo, updated the code!